src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java

```diff
@@ -93,9 +94,21 @@ private StandardCredentials getCredentials(String credentials) {
 
     public KubernetesClient createClient() throws NoSuchAlgorithmException, UnrecoverableKeyException,
             KeyStoreException, IOException, CertificateEncodingException {
-        ConfigBuilder builder = new ConfigBuilder().withMasterUrl(serviceAddress)
-                .withRequestTimeout(readTimeout * 1000)
-                .withConnectionTimeout(connectTimeout * 1000);
+
+        ConfigBuilder builder;
+        // autoconfigure if url is not set
+        if (StringUtils.isBlank(serviceAddress)) {
+            LOGGER.log(FINE, "Autoconfiguring Kubernetes client");
+            builder = new ConfigBuilder(Config.autoConfigure());
+        } else {
+            // although this will still autoconfigure based on Config constructor notes
+            // In future releases (2.4.x) the public constructor will be empty.
+            // The current functionality will be provided by autoConfigure().
+            // This is a necessary change to allow us distinguish between auto configured values and builder values.
```

Comment:
I have to admit that I don't fully understand this comment.

I guess it means that by 2.4.x, this

```
new ConfigBuilder().withMasterUrl(serviceAddress)
```

Will behave differently, i.e. if we move the dependency to 2.4.x we will either introduce a breaking change, or we will have to change that line again?

Reply:
it comes from https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L161

```
  //In future releases (2.4.x) the public constructor will be empty.
  //The current functionality will be provided by autoConfigure().
  //This is a necessary change to allow us distinguish between auto configured values and builder values.
```

Reply:
But this will still cause a breaking change if somebody changes the client library to 2.4 and doesn't update this code here, right?

* before 2.4.x, providing a master url means: do autoconfigure for everything *but* the master url
* after 2.4.x, providing a master url will mean "do not autoconfigure at all".

Is that intended?

Reply:
yes, it may be a breaking change to document. Better than the current behavior because your jenkins settings may be ignored
btw client is in 2.6.x and code still there

Reply:
> Better than the current behavior because your jenkins settings may be ignored

Right, I remember we had that problem as well.

The only remaining downside that remains is that as of now, you can't have autoconfigure AND overwrite the kubernetes URL, like it used to work.

One solution for this would be to add a checkbox for autoconfigure to make it explicit...